clarity/evaluator/haspi/haspi.py

```diff
@@ -165,20 +165,29 @@ def haspi_v2_be(  # pylint: disable=too-many-arguments
 
     # Adjust listener.audiogram_levels_l and _r to match the frequencies above
     hearing_loss_left = [
-        audiogram_left[i] for i in range(len(audiogram_cfs)) if audiogram_cfs[i] in aud
+        audiogram_left[i]
+        for i in range(len(audiogram_frequencies))
+        if audiogram_frequencies[i] in aud
     ]
```

Comment:
I am wondering if the scores should be responsible for this validation. In fact, the only thing this is doing is just filtering the frequencies if the `audiogram_frequencies` are in `aud` but there is no validation that all the `hearing_loss_left` has an audiogram value for each frequency in `aud`

I would consider later making the user responsible for having an audiogram for each frequency and the score validates that all the `audiogram_frequencies` are included.

Reply:
Yes. agree. For now, I will at least fix it so that it throws a ValueError if a needed frequency is missing. This will look different when audiogram objects are used anyway.